### ISSUE TITLE
Context.Request.Type being mapped to ActivityTypes class is misleading - suggest RequestTypes (the pattern we used in v3)

### ISSUE THREAD

**Issue by brandonh-msft** [Microsoft/botbuilder-dotnet#116]
When switching on request types coming in from the user, developers would like to use constants/enums to get type safety and avoid fat-fingering errors.

That said, in BBv3 we had "ActivityTypes" class full of string constants to use for this purpose. In v4, we have `context.*Request*.Type` yet no accompanying `RequestTypes` class with the constants to be used for comparison. Users are left guessing if `ActivityTypes` really is the thing they need, or if they simply need to debug, see what comes through, and switch on that.

**Comment by cleemullins** [comment]
I *think* in the proposed sample that John & I pushed earlier today, we have addressed that. 

If you look in the [BotController](https://github.com/Microsoft/botbuilder-dotnet/blob/master/samples/Microsoft.Bot.Samples.Simplified.Asp/BotController.cs) that was added today to simplify ASP.Net development, we actually created Virtual Methods that developers could override, and each of the Virtual Methods has the already-casted Activity sent in.

Note: Our goal is to push that class down into the Framework (once we get some eyes on it). Doing so seems like it would greatly simplify development by making Activity handling far more explicit. 

Does this address what you're looking for? Otherwise, as you point out, we can put the ActivityTypes constants in a more easy-to-find place.
